### ISSUE TITLE
kubernetes-helmPlugins.helm-secrets: depends on non-free vault package

### ISSUE THREAD

**Issue by brianmay** [NixOS/nixpkgs#309976]
### Describe the bug

Updating my falke to latest nixpkgs unstable gives an error that vault is non-free.

But my package doesn't use or need vault.

My package at https://github.com/electronicarts/helmci/ has:

```nix
        helm = pkgs.wrapHelm pkgs.kubernetes-helm {
          plugins = [
            pkgs.kubernetes-helmPlugins.helm-diff
            pkgs.kubernetes-helmPlugins.helm-secrets
          ];
          extraMakeWrapperArgs =
            "--set HELM_SECRETS_SOPS_PATH ${sops}/bin/sops --set HELM_SECRETS_VALS_PATH ${vals}/bin/vals";
        };
```

This gives the error:

```
$ nix build
warning: Git tree '/home/brian/tree/ea/cloudcell/helmci' is dirty
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'helmci-all'
         whose name attribute is located at /nix/store/3pif36ks3f56py4wb1dkq6sh0nkf3ygj-source/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'buildCommand' of derivation 'helmci-all'

         at /nix/store/3pif36ks3f56py4wb1dkq6sh0nkf3ygj-source/pkgs/build-support/trivial-builders/default.nix:68:16:

           67|         enableParallelBuilding = true;
           68|         inherit buildCommand name;
             |                ^
           69|         passAsFile = [ "buildCommand" ]

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: Package ‘vault-1.16.2’ in /nix/store/3pif36ks3f56py4wb1dkq6sh0nkf3ygj-source/pkgs/tools/security/vault/default.nix:47 has an unfree license (‘bsl11’), refusing to evaluate.

       a) To temporarily allow unfree packages, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNFREE=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnfree = true; }
       in configuration.nix to override this.

       Alternatively you can configure a predicate to allow specific packages:
         { nixpkgs.config.allowUnfreePredicate = pkg: builtins.elem (lib.getName pkg) [
             "vault"
           ];
         }

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnfree = true; }
       to ~/.config/nixpkgs/config.nix.
 ```

If I comment out the `helm-secrets` it works.

### Steps To Reproduce
1. `git clone https://github.com/electronicarts/helmci.git`
2. `cd helmci`
3. Modify `flake.nix` to use nixpkgs-unstable instead of nixpkgs 23.11
4. `nix flake update`
5. `nix build`

### Expected behavior
Should build without errors.

### Screenshots
As above.

### Notify maintainers

@yurrriq

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.81, NixOS, 23.11 (Tapir), 23.11.20240308.2be119a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(brian): `""`
 - channels(root): `"nixos-23.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos````

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by teto** [comment]
helm-secrets depends on vault

**Comment by brianmay** [comment]
What happened to vault? It use to be open source I thought. Did the license change?

**Comment by brianmay** [comment]
Does helm-secrets really depend on vault? From the About section it says:

> Store your secrets a cloud native secret manager like AWS SecretManager, Azure KeyVault or HashiCorp Vault and inject them inside value files or templates.

Which seems to indicate that it should be able to use any secret manager, not just vault.

**Comment by teto** [comment]
the vault license change, which sparked quite the outrage.

> Does helm-secrets really depend on vault? From the About section ...

you ask a question and follow with the answer, that's both rare and impressive ^^'' 
You can see it here:
https://github.com/NixOS/nixpkgs/blob/a10842c7f0343bad733d039c72cf15a6ca0d1eb4/pkgs/applications/networking/cluster/helm/plugins/helm-secrets.nix#L32
I suggest you follow the recommendation and accept the new license for vault. If you dont need it, you can remove support for it as well from helm-secrets (but will require some nix skills though you can search the web for plenty of examples)

**Comment by brianmay** [comment]
How about switching the dependency from vault to OpenBAO - would that be possible?

**Comment by yurrriq** [comment]
> How about switching the dependency from vault to OpenBAO - would that be possible?

Heh, I didn't realize LF forked Vault too. :popcorn:

It's certainly possible, but might have unintended side effects for Nix users. (I realize how annoying that might sound, given your recent experience.)

By default, helm-secrets supports [vals](https://github.com/helmfile/vals) and [sops](https://github.com/getsops/sops). After a quick glance, it looks like both sops and vals use the Vault Go library, so maybe we don't even need `vault` at runtime anyway...

I wonder if sops and vals will have to alter their licenses...

**Comment by yurrriq** [comment]
I've just tried a modified helm-secrets derivation, removing all references to `vault`, against a repo that uses helm-secrets (with AWS KMS backend) and it worked fine. I then switched the config to point use a Vault backend, and that also worked. Note that I used `nix run nixpkgs-unfree#vault login` to get a token, but assuming `~/.vault-token` exists or the env var `VAULT_TOKEN` is set somehow, it seems `helm-secrets` doesn't need a `vault` binary at runtime at all. So, I suggest we drop the dependency.
